File: src/main/java/io/coconut/ferrum/VersionType.java
package io.coconut.ferrum;

public enum VersionType {
    RELEASE("release"),
    SNAPSHOT("snapshot"),
    BETA("old_beta"),
    ALPHA("old_alpha");

    private final String friendlyName;

    VersionType(String friendlyName) {
        this.friendlyName = friendlyName;
    }

    public String getFriendlyName() {
        return friendlyName;
    }
}

File: src/io/coconut/ferrum/types/VersionType.scala
package io.coconut.ferrum.types

enum VersionType(val secondName: String) {
  case RELEASE extends VersionType("release")
  case SNAPSHOT extends VersionType("snapshot")
  case BETA extends VersionType("old_beta")
  case ALPHA extends VersionType("old_alpha")

  // Fallback for Java
  def getSecondName: String = secondName
}
